server.go

```diff
@@ -838,29 +845,46 @@ func (s *Server) syncLocalOrderState() error {
 // determineBatchVersion determines the batch version that will be sent to the
 // auctioneer to signal compatibility with the different features the trader
 // client can support.
-func (s *Server) determineBatchVersion() order.BatchVersion {
+func (s *Server) determineBatchVersion() (order.BatchVersion, error) {
 	configVersion := s.cfg.DebugConfig.BatchVersion
 
 	// We set the default value of the config flag to -1, so we can
 	// differentiate between no value set and the first version (0).
 	if configVersion >= 0 {
-		return order.BatchVersion(configVersion)
+		return order.BatchVersion(configVersion), nil
 	}
 
-	isTaprootCompatible := false
-	verErr := lndclient.AssertVersionCompatible(
-		s.lndServices.Version, taprootVersion,
+	ctxt, cancel := context.WithTimeout(
+		context.Background(), time.Second*10,
 	)
-	if verErr == nil {
-		isTaprootCompatible = true
+	defer cancel()
+
+	// We will try to decided what version to used based on the latest
+	// batch version + what features the node supports.
+	info, err := s.lndClient.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+	if err != nil {
+		return 0, err
+	}
+
+	// If the node supports ZeroConfChannels use that batch version.
+	_, supportsZeroConf := info.Features[uint32(lnwire.ZeroConfOptional)]
```

Comment:
Should we plan ahead and check if either the `lnwire.ZeroConfReqired` or `lnwire.ZeroConfOptional` is turned on? And same for zero conf?

Reply:
Solved... i think